nixos/modules/misc/ids.nix

```diff
@@ -347,6 +347,7 @@ in
       #mailman = 316;  # removed 2019-08-30
       zigbee2mqtt = 317;
       # shadow = 318; # unused
+      calibre-web = 319;
```

Comment:
Aren't hard-coded uids for services supposed to stop being used?

Reply:
Yes, you are right. Good catch!

Reply:
> Aren't hard-coded uids for services supposed to stop being used?

@SuperSandro2000 Ok, will remove. What is the correct way to specify user/group in this case? `calibre-web` needs r/w access to Calibre DB. What should I do in this case? Should I use systemd's `DynamicUser` and add provide some `SupplimentaryGroups` that have r/w access to the DB?

Reply:
Can't you just not specify an ID for the `calibre-web` user and group. NixOS should pick an ID automatically.

You might need to specify that it is a system user.

Reply:
See this paragraph from the linked RFC.

> Otherwise let NixOS assign dynamic persistent ids, which happens automatically when users.users.<name?>.uid/users.groups.<name?>.gid is not set. For users, users.users.<name?>.isSystemUser should be set so that only uids under 1000 are used by NixOS services.

Reply:
> Can't you just not specify an ID for the `calibre-web` user and group. NixOS should pick an ID automatically.

Sure I can. But this automatically generated user won't have write permissions to the original Calibre DB.